sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -597,6 +597,19 @@ async def get_throughput(self, **kwargs: Any) -> ThroughputProperties:
         if response_hook:
             response_hook(self.client_connection.last_response_headers, throughput_properties)
 
+        if 'offerAutopilotSettings' in throughput_properties[0]['content'] and 'autoUpgradePolicy' in \
+                throughput_properties[0]['content']['offerAutopilotSettings']:
+            return ThroughputProperties(properties=throughput_properties[0], auto_scale_max_throughput= \
+                throughput_properties[0]['content']['offerAutopilotSettings']['maxThroughput'],
+                                        auto_scale_increment_percent=throughput_properties[0]['content']
+                                        ['offerAutopilotSettings']['autoUpgradePolicy']['throughputPolicy'][
+                                            'incrementPercent'])
+
+        elif 'offerAutopilotSettings' in throughput_properties[0]['content']:
+            return ThroughputProperties(properties=throughput_properties[0],
+                                        auto_scale_max_throughput=throughput_properties[0]['content']
+                                        ['offerAutopilotSettings']['maxThroughput'])
+
```

Comment:
Why is this conditional logic necessary? It seems like this has a high chance of silently ignoring things.

Reply:
This logic is used for deserializing the auto scale properties. We cannot just return them all at once as we need to check if the container or database has autoscale. If we tried to return all the properties then it would fail because if we tried to use a container with no autoscale it would tell us that some of the properties we are returning are not found.

Reply:
I have really tested this out plenty and this doesn't ignore anything that I am aware of and everything works how it should.